.github/workflows/tests.yml

```
@@ -35,60 +35,45 @@ jobs:
+      - name: Run tests for HTTP Mode adapters
         run: |
-          pytest tests/adapter_tests/aws/
-      - name: Run tests for HTTP Mode adapters (Bottle)
-        run: |
-          pytest tests/adapter_tests/bottle/
-      - name: Run tests for HTTP Mode adapters (CherryPy)
-        run: |
-          pytest tests/adapter_tests/cherrypy/
-      - name: Run tests for HTTP Mode adapters (Django)
-        run: |
-          pytest tests/adapter_tests/django/
-      - name: Run tests for HTTP Mode adapters (Falcon)
-        run: |
-          pytest tests/adapter_tests/falcon/
-      - name: Run tests for HTTP Mode adapters (Flask)
-        run: |
-          pytest tests/adapter_tests/flask/
-      - name: Run tests for HTTP Mode adapters (Pyramid)
-        run: |
-          pytest tests/adapter_tests/pyramid/
-      - name: Run tests for HTTP Mode adapters (Starlette)
-        run: |
-          pytest tests/adapter_tests/starlette/
-      - name: Run tests for HTTP Mode adapters (Tornado)
-        run: |
-          pytest tests/adapter_tests/tornado/
-      - name: Run tests for HTTP Mode adapters (WSGI)
-        run: |
-          pytest tests/adapter_tests/wsgi/
+          pytest tests/adapter_tests/ \
+          --ignore=tests/adapter_tests/socket_mode/ \
+          --ignore=tests/adapter_tests/asgi/ \
+          --junitxml=reports/test_adapter.xml
       - name: Install async dependencies
         run: |
           pip install -r requirements/async.txt
+      - name: Run tests for HTTP Mode adapters (ASGI)
+        run: |
+          # Requires async test dependencies
+          pytest tests/adapter_tests/asgi/ --junitxml=reports/test_adapter_asgi.xml
       - name: Run tests for Socket Mode adapters
         run: |
           # Requires async test dependencies
-          pytest tests/adapter_tests/socket_mode/
+          pytest tests/adapter_tests/socket_mode/ --junitxml=reports/test_adapter_socket_mode.xml
       - name: Run tests for HTTP Mode adapters (asyncio-based libraries)
         run: |
-          pytest tests/adapter_tests_async/
-      - name: Run tests for HTTP Mode adapters (ASGI)
-        run: |
-          # Requires async test dependencies
-          pytest tests/adapter_tests/asgi/
+          pytest tests/adapter_tests_async/ --junitxml=reports/test_adapter_async.xml
       - name: Install all dependencies
         run: |
           pip install -r requirements/testing.txt
       - name: Run asynchronous tests
         run: |
-          pytest tests/slack_bolt_async/
-          pytest tests/scenario_tests_async/
+          pytest tests/slack_bolt_async/ --junitxml=reports/test_slack_bolt_async.xml
+          pytest tests/scenario_tests_async/ --junitxml=reports/test_scenario_async.xml
+      - name: Upload test results to Codecov
+        if: ${{ !cancelled() }}
+        uses: codecov/test-results-action@v1
+        with:
+          directory: ./reports/
+          files: test*.xml
```

```suggestion
          files: ./**/test*.xml
```

https://github.com/slackapi/bolt-python/actions/runs/14134899037/job/39604145700#step:14:183

Right now this raises a warning in outputs 🔍 

```
Some files were not found --- {"not_found_files": ["test*.xml"]}
```

I think a standalone `directory` option might be another option, and this isn't a blocker at all! Just wanting to share findings ✨

---

Thats a great catch 🥇  I totally missed that warning, I'll try testing a few things out